Sources/BluemixAppID/APIKituraCredentialsPluginConfig.swift

```diff
@@ -0,0 +1,77 @@
+/*
+ Copyright 2017 IBM Corp.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+
+import Foundation
+import SwiftyJSON
+import SimpleLogger
+
+internal class APIKituraCredentialsPluginConfig {
+    private let vcapServices = "VCAP_SERVICES"
+    private let vcapServicesCredentials = "credentials"
+    private let vcapServicesName = "AdvancedMobileAccess"
+    private let vcapApplication = "VCAP_APPLICATION"
+    private let oauthServerURL = "oauthServerUrl"
+    private let pubkeyServerURL = "pubKeyServerUrl"
+    private let logger = Logger(forName: "APIKituraCredentialsPluginConfig")
+    var serviceConfig: [String:Any] = [:]
+    public init(options:[String:Any]?) {
+        logger.debug("Intializing APIKituraCredentialsPluginConfig")
+        let options = options ?? [:]
+        let vcapString = ProcessInfo.processInfo.environment[self.vcapServices] ?? ""
+        let vcapServices = JSON.parse(string: vcapString)
+        var vcapServiceCredentials: [String:Any]? = [:]
+        if vcapServices.dictionary != nil {
+            for (key,value) in vcapServices.dictionary! {
```

Comment:
and then look for the credentials:

// Find App ID service config
	for (var propName in vcapServices) {
		// Does service name starts with VCAP_SERVICES_SERVICE_NAME
		if (propName.indexOf(VCAP_SERVICES_SERVICE_NAME1) === 0||propName.indexOf(VCAP_SERVICES_SERVICE_NAME2) === 0) {
			vcapServiceCredentials = vcapServices[propName][0][VCAP_SERVICES_CREDENTIALS];
			break;
		}
	}

-----------
This is to deal with service rename that happened some time ago, so we could have users with different VCAP settings. Now it's AppID, so I think you can remove AdvancedMobileAccess at all, or just adapt the logic above.

Reply:
Thanks for the review!
Addressed in commit f77ed01.